src/game.py

```diff
@@ -0,0 +1,227 @@
+# make_sudoku.py
+# author: David Bau
+# Copyright (c) 2006 David Bau.  All rights reserved.
+# URL: http://davidbau.com/downloads/sudoku.py
+# Modified by Daniel Jauregui for study purposes
+
+import random
+
+
+class Game(object):
+
+    def __init__(self, level):
+        self.level = level
+
+    def generate_game(self):
+        """Call to make_puzzle with a empty board to generate game
+        with level defined and it will return a array e.g.: Array: [4, 9, 3, None, 5, ...., 8, 8, None, None, 1]
+        Where the position are from 0 to 80
+
+        return -- Array e.g.: [4, 9, 3, None, 5, ...., 8, 8, None, None, 1]
+        """
+        return self.make_puzzle(self.solution([None] * 81))
+
+    def make_puzzle(self, board):
+        """Get the solved board and remove the range of values
+         defined for level, the random position values will be
+         replaced for None.
+
+        Keyword arguments:
+        board -- Array with all values solved for sudoku.
+        return -- Array e.g.: [4, 9, 3, None, 5, ...., 8, 8, None, None, 1]
+        """
+        """ Implement random according level defined """
+        puzzle = []
+        # Deduced will use basic board for iteration conditional and fill it for each value found
+        deduced = [None] * 81
+        order = random.sample(xrange(81), 81)
+        for position in order:
+            if deduced[position] is None:
+                puzzle.append((position, board[position]))
+                deduced[position] = board[position]
+                self.deduce(deduced)
+        return self.board_for_entries(puzzle)
+
+    def board_for_entries(self, entries):
+        """Get array with tuples of position and value.
+        This method will return an array with final array
+        with values (Numbers and None).
+
+        Keyword arguments:
+        entries -- Array of tuples e.g.: [(23, 5), (80, 9),..., (2, 5), (15, 1)]
+        return  -- Array e.g.: [4, 9, 3, None, 5, ...., 8, 8, None, None, 1]
+        """
+        board = [None] * 81
+        for position, number in entries:
+            board[position] = number
+        return board
+
+    def solution(self, board):
```

Comment:
remember that method names should be actions, like get_solution, solution is a noun